p2p/common/error.go

```diff
@@ -0,0 +1,18 @@
+package common
+
+import "fmt"
+
+type ErrFactory func(msg string, err error) error
+
+func NewErrFactory(preMsg string) ErrFactory {
```

Comment:
This is okay, but not a fan of introducing new patterns in just one specific module.

Would need one of the following:
1. TODO explaining we should adopt this pattern everywhere
2. A GitHub ticket or Github discussion where we do this
3. Just going with our general pattern for now:

```go
func ErrUnknownEventType(msg any) error {
	return fmt.Errorf("unknown event type: %v", msg)
}
```

Reply:
Hehe - my intent was not to introduce a new pattern necessarily. I'm actually not especially proud of this and was hoping to get some feedback about what conventional error handling should look like.

Reply:
I feel like it's worth noting though that it did prove quite useful during debugging to have wrapped those errors in a string which I could search for. :sweat_smile:

Reply:
Okay, I ended up going for option 2. I've added a `DISCUSS` to the code and opened #519.